runtime/bcutil/ClassFileOracle.cpp

```diff
@@ -457,6 +457,12 @@ ClassFileOracle::walkFields()
 			case CFR_ATTRIBUTE_Deprecated:
 				/* Do nothing */
 				break;
+#if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
```

Comment:
`J9VM_OPT_VALHALLA_VALUE_TYPES` used for value object [JEP](https://openjdk.org/jeps/8277163).

The `NullRestricted` is introduced in JEP 401 Flattened Heap Layouts for Value Objects, so the code related to `NullRestricted` attribute should be inside `J9VM_OPT_VALHALLA_FLATTENABLE_VALUE_TYPES`. The same applies to `ImplicitCreation` as well. Can you fix the code dealing with `ImplicitCreation` introduced by https://github.com/eclipse-openj9/openj9/pull/17655 as well ?

Reply:
Sure. So Preload is under `J9VM_OPT_VALHALLA_VALUE_TYPES ` and ImplicitCreation and NullRestricted should be under `J9VM_OPT_VALHALLA_FLATTENABLE_VALUE_TYPES `.

What is the purpose of differentiating these? The jeps are related.

Reply:
> What is the purpose of differentiating these? The jeps are related.

The JEPs are likely to be released in multiple JDK versions. The value object JEP is going to be released first. We need to be able to split them in the code.

Reply:
I've added a commit to correct the compiler flags for all three attributes. However this Preload logic is already under `J9VM_OPT_VALHALLA_FLATTENABLE_VALUE_TYPES ` https://github.com/eclipse-openj9/openj9/blob/master/runtime/vm/createramclass.cpp#L1947-L1960. Is that what you would expect?

Reply:
> Is that what you would expect?

Yes. For the value objects JEP, OpenJ9 needs to recognize the Preload attribute only. Value objects won't be flattened by OpenJ9. We need to do the actually preloading when the fields are flattenable.